app/src/main/java/com/duckduckgo/app/browser/WebViewRequestInterceptor.kt

```diff
@@ -106,6 +126,31 @@ class WebViewRequestInterceptor(
         return null
     }
 
+    private suspend fun newUserAgent(
+        request: WebResourceRequest,
+        webView: WebView,
+        webViewClientListener: WebViewClientListener?
+    ): String? {
+        return if (request.isForMainFrame && request.method == "GET") {
+            val url = request.url ?: return null
+            val isOldPage = withContext(Dispatchers.Main) { (webView.copyBackForwardList().currentIndex < webView.copyBackForwardList().size - 1) }
+            if (isOldPage) return null
+            val desktopSiteEnabled = webViewClientListener?.isDesktopSiteEnabled() == true
+            val currentAgent = withContext(Dispatchers.Main) { webView.settings?.userAgentString }
```

Comment:
Just double-checking that we need to call this method from Main thread.

Reply:
Yeah that is correct.